cmd/geth/main.go

```diff
@@ -346,6 +350,20 @@ func main() {
 	}
 }
 
+func makeDefaultExtra() []byte {
+	var clientInfo = struct {
+		Version   uint
+		Name      string
+		GoVersion string
+		Os        string
+	}{uint(VersionMajor<<16 | VersionMinor<<8 | VersionPatch), ClientIdentifier, runtime.Version(), runtime.GOOS}
+	extra, err := rlp.EncodeToBytes(clientInfo)
```

Comment:
While this works now, we have no control over values returned from runtime, and while unlikely it's plausible they could return longer strings causing the RLP to be larger than 32 bytes, breaking mining for someone upgrading Go version.

Add a check to truncate strings to guarantee RLP is never more than 32 bytes.